Mathlib/Geometry/Manifold/Instances/Real.lean

```diff
@@ -90,6 +90,26 @@ theorem range_euclideanHalfSpace (n : ℕ) [Zero (Fin n)] :
   Subtype.range_val
 @[deprecated (since := "2024-04-05")] alias range_half_space := range_euclideanHalfSpace
 
+-- TODO: generalise these lemmas to other values of `p`
```

Comment:
Why not do this now? I think it's relatively easy (note: I also don't like the `change` utilizing the defeq there).
```lean
open ENNReal in
theorem interior_halfspace {a : ℝ} {n : ℕ} {p : ℝ≥0∞} (i : Fin n) :
    interior { y : PiLp p (fun _ : Fin n ↦ ℝ) | a ≤ y i } = { y | a < y i } := by
  let f : PiLp p (fun _ : Fin n ↦ ℝ) →L[ℝ] ℝ := ContinuousLinearMap.proj i
  simpa [interior_Ici] using f.interior_preimage (Function.surjective_eval _) (Ici a)
```
This allows your lemma as a corollary, even using `with_reducible` since `EuclideanSpace` is an `abbrev`
```lean
example {a : ℝ} {n : ℕ} (i : Fin n) :
    interior { y : EuclideanSpace ℝ (Fin n) | a ≤ y i } = { y | a < y i } := by
  with_reducible exact interior_halfspace i
```
Finally, I'm not sure I understand the choices behind your use of explicit / implicit variables here. Why is `i` explicit but `a` and `n` implicit?

Reply:
I was thinking of allowing different component spaces... in which case I'd even need to think about the statement.
I like your generalisation and proof, thanks!

I had simply overlooked that i could be implicit as well, and just made it so.